server/internal/usecase/pageinfo.go

```diff
@@ -0,0 +1,63 @@
+package usecase
+
+type PageInfo struct {
+	totalCount      int
+	startCursor     *Cursor
+	endCursor       *Cursor
+	hasNextPage     bool
+	hasPreviousPage bool
+}
```

Comment:
```suggestion
type PageInfo struct {
	TotalCount      int
	StartCursor     *Cursor
	EndCursor       *Cursor
	HasNextPage     bool
	HasPreviousPage bool
}
```

This struct contains very little domain logic, so please make all fields public, remove all NewPageInfo functions and methods, and move it to pagination.go.